gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -259,6 +261,15 @@ public void commit()
       throws IOException {
     closeInternal();
     super.commit();
```

Comment:
This line calls `FsDataWriter` and triggers the moving from staging to output directory. Is there a reason for us to do all that work and then do the validation? 

I also wonder why this is part of the commit step and not part of the close step. close does not call this method, but it does do the flush.

If we close and the flushed file turns out not to be valid, we will miss the validation here.

Reply:
I was thinking maybe there's something wrong during moving. But given the issue is malformed files, so the issue should already be there after writer closed. So move the logic to after closeInternal() is called.